bundles/binding/org.openhab.binding.networkupstools/src/main/java/org/openhab/binding/networkupstools/internal/NetworkUpsToolsBinding.java

```diff
@@ -0,0 +1,209 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.networkupstools.internal;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.networkupstools.jnut.Client;
+import org.networkupstools.jnut.Device;
+import org.networkupstools.jnut.Variable;
+import org.openhab.binding.networkupstools.NetworkUpsToolsBindingProvider;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+	
+
+/**
+ * Implement this class if you are going create an actively polling service
+ * like querying a Website/Device.
+ * 
+ * @author jaroslawmazgaj
+ * @since 1.7.0
+ */
+public class NetworkUpsToolsBinding extends AbstractActiveBinding<NetworkUpsToolsBindingProvider> implements ManagedService {
+
+	private static final Logger logger = 
+		LoggerFactory.getLogger(NetworkUpsToolsBinding.class);
+
+	private Map<String, NutConfig> upses = new HashMap<String, NutConfig>(); 
+	
+	/** 
+	 * the refresh interval which is used to poll values from the NetworkUpsTools
+	 * server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+	
+		
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "NetworkUpsTools Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		Multimap<String, ItemDefinition> items = HashMultimap.create();
+		for (NetworkUpsToolsBindingProvider provider : providers) {
+			for (String itemName : provider.getItemNames()) {
+				String name = provider.getUps(itemName);
+				Class<? extends Item> itemType = provider.getItemType(itemName);
+				String property = provider.getProperty(itemName);
+				items.put(name, new ItemDefinition(itemName, itemType, property));
+			}
+		}
+		
+		for(String name : items.keySet()) {
+			NutConfig nut = upses.get(name);
+			if (nut == null) {
+				logger.error("No configuration for UPS with name: '{}'", name); 
```

Comment:
you should add a "return" otherwise you will get NPEs later on because nut IS null then …